Nucleation_model_histogram.py

```diff
@@ -87,18 +87,14 @@ def residual(t, sol_vec):
             Dnp_dt[i] += DN_Dt
             break
     for i, N in enumerate(n_p):
-        if N == 0:
-            Dr_dt[i] = 0
-            Dnp_dt[i] += 0
-        else:
-            Dr_dt[i] = D_LiO2*V*(C_Li- C_Li_sat)*(C_LiO2-C_LiO2_sat)/(radii[i]+D_LiO2/k_surf)*1E5- m.pi*radii[i]**2*N*gamma_surf*k_surf_des
-            dNdt_radii = Dr_dt[i]/bin_width*n_p[i]
-            if dNdt_radii <0 and i > 0:
-                Dnp_dt[i] += dNdt_radii
-                Dnp_dt[i-1] -= dNdt_radii
-            elif dNdt_radii > 0 and radii[i] != radii[-1]:
-                Dnp_dt[i] -= dNdt_radii
-                Dnp_dt[i+1] += dNdt_radii
+        Dr_dt[i] = D_LiO2*V*(C_Li- C_Li_sat)*(C_LiO2-C_LiO2_sat)/(radii[i]+D_LiO2/k_surf)*1E5- m.pi*radii[i]**2*N*gamma_surf*k_surf_des
+        dNdt_radii = Dr_dt[i]/bin_width*n_p[i]
+        if dNdt_radii <0 and i > 0:
```

Comment:
I suspect that there needs to be a third branch to this conditional, for `i=0`, or perhaps a nested `if`.  If `i=0` and `dNdt_radii <0`, you still want the `Dnp_dt[i] += dNdt_radii` line, just not the `Dnp_dt[i-1]`.  So maybe something like
```suggestion
        if dNdt_radii <0:
            Dnp_dt[i] += dNdt_radii
            if i > 0:
                Dnp_dt[i-1] -= dNdt_radii
```